signer/tests/integration/setup.rs

```diff
@@ -440,6 +415,27 @@ pub async fn backfill_bitcoin_blocks(db: &PgStore, rpc: &Client, chain_tip: &bit
     }
 }
 
+/// Fetch all block headers from bitcoin-core and store it in the database.
+pub async fn fetch_canonical_bitcoin_blockchain(db: &PgStore, rpc: &Client) {
```

Comment:
maybe nit: given the usual usage we do of this we could also return `db.get_bitcoin_canonical_chain_tip().await.unwrap().unwrap()` for ergonomics

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/1406/commits/c7d18638e9733d402d3c47adfeb3c68351f49001.

Reply:
`chain_tip_info.best_block_hash.into()` may not be the db chain tip, returning it may be confusing; or maybe not, feel free to resolve in that case

Reply:
Maybe I should add an assert

Reply:
Couldn't hurt, so I added it in https://github.com/stacks-network/sbtc/pull/1406/commits/efc0622d2e19c8dd33d1c25e5bb53fe3c7d5cf3a.